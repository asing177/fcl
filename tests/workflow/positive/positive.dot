digraph workflow {
edge [color="#000000"]
graph [bgcolor=transparent]


initial [shape=point; width=0.3; style=filled; color="#0e64ce"; label=""]
a [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
b [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
c [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
d [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
e [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
f [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
terminal [shape=point; width=0.3; peripheries=2; style=filled; color="#d11010"; label=""]

1[label=<<FONT POINT-SIZE="16">toA</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
2[label=<<FONT POINT-SIZE="16">toB</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
3[label=<<FONT POINT-SIZE="16">toB</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
4[label=<<FONT POINT-SIZE="16">toCorD</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
5[label=<<FONT POINT-SIZE="16">toCorD</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
6[label=<<FONT POINT-SIZE="16">toEorF</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
7[label=<<FONT POINT-SIZE="16">toEorF</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
8[label=<<FONT POINT-SIZE="16">toEfromD</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
9[label=<<FONT POINT-SIZE="16">toEfromF</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
10[label=<<FONT POINT-SIZE="16">finish</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]

initial -> 1 ; 1 -> a
a -> 2 ; 2 -> a
a -> 3 ; 3 -> b
b -> 4 ; 4 -> c
b -> 5 ; 5 -> d
c -> 6 ; 6 -> e
c -> 7 ; 7 -> f
d -> 8 ; 8 -> e
f -> 9 ; 9 -> e
e -> 10 ; 10 -> terminal



}
