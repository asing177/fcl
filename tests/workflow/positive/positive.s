// Everything good in this test.

transition initial -> a;
transition a -> a;
transition a -> b;
transition b -> c;
transition b -> d;
transition c -> e;
transition c -> f;
transition d -> e;
transition f -> e;
transition e -> terminal;

@initial
toA() {
  transitionTo(:a);
}

@a
toB() {
  myDate = "2018-01-22T13:46:10+00:00";

  if ((myDate < now()) && (now() < myDate + 7d)) {
    transitionTo(:b);
  } else {
    stay();
  };
}

@b
toCorD() {
  if (now() < "2018-01-22T13:46:10+00:00") {
    transitionTo(:c);
  } else {
    transitionTo(:d);
  };
}

@c
toEorF() {
  if (true) {
    transitionTo(:e);
  } else {
    transitionTo(:f);
  };
}

@d
toEfromD() {
  transitionTo(:e);
}

@f
toEfromF() {
  transitionTo(:e);
}

@e
finish() {
  terminate();
}
