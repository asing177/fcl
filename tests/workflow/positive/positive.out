 
Method arguments and effects:
toA: () -> {}
toB: () -> {write via stay, read via now}
toCorD: () -> {read via now}
toEorF: () -> {}
toEfromD: () -> {}
toEfromF: () -> {}
finish: () -> {write via terminate}